src/internal/parser.ts

```diff
@@ -76,44 +76,50 @@ export class Parser {
       scope?: string;
     }
   ): Promise<ProfileDocumentNode> {
-    const hash = Parser.hash(input);
-    const hashedName = `${info.profileName}-${hash}${EXTENSIONS.profile.build}`;
+    const sourceChecksum = new Source(input, fileName).checksum();
     const cachePath = joinPath(
       Config.instance().cachePath,
       ...[...(info.scope !== undefined ? [info.scope] : [])]
     );
-    const hashedPath = joinPath(cachePath, hashedName);
+    const path = joinPath(
+      cachePath,
+      `${info.profileName}${EXTENSIONS.profile.build}`
+    );
 
     // If we have it in memory cache, just return it
-    if (this.profileCache[hashedPath] !== undefined) {
-      return this.profileCache[hashedPath];
+    if (
+      this.profileCache[path] !== undefined &&
+      this.profileCache[path].astMetadata.sourceChecksum === sourceChecksum
+    ) {
+      return this.profileCache[path];
     }
 
-    // If we already have parsed map in cache file, load it
-    {
-      const parsedProfile = await Parser.loadCached(
-        hashedPath,
-        assertProfileDocumentNode,
-        this.profileCache
-      );
-      if (parsedProfile !== undefined) {
-        return parsedProfile;
-      }
+    // If we already have valid AST in cache file, load it
+    let parsedProfile = await Parser.loadCached(
+      path,
+      isProfileDocumentNode,
+      this.profileCache,
+      sourceChecksum
+    );
+    // If we have cached AST, we can use it.
+    if (parsedProfile !== undefined) {
+      return parsedProfile;
     }
 
     // If not, delete old parsed profiles
-    await Parser.clearFileCache(
-      `${info.profileName}-[0-9a-f]+${EXTENSIONS.profile.build}`,
-      cachePath
-    );
+    await Parser.clearFileCache(path);
 
     // And write parsed file to cache
-    const parsedProfile = parseProfile(new Source(input, fileName));
+    parsedProfile = parseProfile(new Source(input, fileName));
+    if (!isProfileDocumentNode(parsedProfile)) {
+      //TODO: more helpful error - can this be product of not matching AST package versions?
```

Comment:
yes, do

Reply:
FIxed in: https://github.com/superfaceai/one-sdk-js/pull/166/commits/120f1b8f355ae6e288b8e2f3aaecdd8b1ca2bb2c